contracts/src/v0.8/functions/dev/v1_0_0/FunctionsBilling.sol

```diff
@@ -218,26 +211,29 @@ abstract contract FunctionsBilling is Routable, IFunctionsBilling {
       callbackGasLimit: request.callbackGasLimit,
       estimatedTotalCostJuels: estimatedTotalCostJuels,
       timeoutTimestamp: uint32(block.timestamp + config.requestTimeoutSeconds),
-      requestId: requestId,
+      requestId: 0x0000000000000000000000000000000000000000000000000000000000000000,
       donFee: donFee,
       gasOverheadBeforeCallback: config.gasOverheadBeforeCallback,
       gasOverheadAfterCallback: config.gasOverheadAfterCallback
     });
 
-    s_requestCommitments[requestId] = keccak256(abi.encode(commitment));
+    commitment.requestId = _computeRequestId(
+      commitment,
+      request.initiatedRequests + 1
+    );
+
+    s_requestCommitments[commitment.requestId] = keccak256(abi.encode(commitment));
 
     return commitment;
   }
 
   /// @notice Generate a keccak hash request ID
   /// @dev uses the number of requests that the consumer of a subscription has sent as a nonce
   function _computeRequestId(
-    address don,
-    address client,
-    uint64 subscriptionId,
+    FunctionsResponse.Commitment memory commitmentWithoutRequestId,
     uint64 nonce
   ) private pure returns (bytes32) {
-    return keccak256(abi.encode(don, client, subscriptionId, nonce));
+    return keccak256(abi.encode(commitmentWithoutRequestId, nonce));
```

Comment:
Is there any meaningful gas overhead of hashing the whole struct?

Reply:
Actually... There shouldn't need to be any changes to the gas overhead since this computation only occurs during the request transaction.  We never use `_computeRequestId` anywhere else (I checked).